Cartesian control

beppe712 opened the pull request:
This is the Cartesian Control API wrapper

GreatAlexander committed:
Changes Header indentation to 1-space tab convention, minor cpp indentation fix

GreatAlexander committed:
Renames motion wrapper to stiffness control, fixes dependencies

beppe712 committed:
first implementation of cartesian control API

beppe712 committed:
Fix commit
  -solved issues which prevented previous commit to compile

beppe712 committed:
Previous commit did not add the files

svepe commented:
There are some binary files which we definitely do not want in the repository.
They all start with ._ so it should be easy to add them to the gitignore file.

beppe712 commented:
I have set up a global gitignore file such that all files generated by Mac OS and binary files are ignored. I have cleaned my cache and added the files again.

beppe712 committed:
Removed all system and binary files

GreatAlexander commented:
Since the original commit you branched of was quite a while ago, and there have been multiple changes to the parent branch in the meantime (for example renaming of motion_wrapper to stiffness) you will need to make sure there are no conflicting changes before doing a pull request.

Right now, Github cannot merge automatically, since it detects we have changed the same files. This means you have to merge locally with the new changes, and very probably deal with merge conflicts.
The whole purpose of us working on separate files is to avoid such issues, as merging is a pain.

My recommendation is to keep working on your branch and we will solve the merge conflicts eventually when we meet. If you are feeling brave, you will first need to read about and download some tool like this:
http://meldmerge.org/

And then, you would need to:
- Commit/Stash any changes you may have done in the meantime on your branch.
- 'git pull origin motion', to get all the new remote changes in the motion branch.
- 'git checkout motion', to move to the branch you want to merge with
- 'git merge cartesian_control', to merge the changes from your branch to the motion branch.
  
  At this point, the computer will scream and complain that it cannot merge automatically, and that there were conflicts. You then use your favourite tool like meld and decide which changes in each conflicting file you would like to keep. When you are done, you should check with me first that all files were merged correctly, and then it can be pushed online!

Good luck!
